metis-pattern-analysis/src/main/java/eu/europeana/patternanalysis/ProblemPatternAnalyzer.java

```diff
@@ -2,37 +2,55 @@
 
 import static java.lang.String.format;
 import static java.util.function.Predicate.not;
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.Collectors.toMap;
+import static java.util.stream.Collectors.toSet;
+import static org.apache.commons.lang3.BooleanUtils.isFalse;
 
 import eu.europeana.metis.schema.convert.RdfConversionUtils;
 import eu.europeana.metis.schema.convert.SerializationException;
-import eu.europeana.metis.schema.jibx.Description;
 import eu.europeana.metis.schema.jibx.EuropeanaType;
 import eu.europeana.metis.schema.jibx.EuropeanaType.Choice;
+import eu.europeana.metis.schema.jibx.LiteralType;
 import eu.europeana.metis.schema.jibx.ProvidedCHOType;
 import eu.europeana.metis.schema.jibx.ProxyType;
 import eu.europeana.metis.schema.jibx.RDF;
-import eu.europeana.metis.schema.jibx.Title;
+import eu.europeana.metis.schema.jibx.ResourceOrLiteralType;
 import eu.europeana.patternanalysis.view.ProblemOccurrence;
 import eu.europeana.patternanalysis.view.ProblemPattern;
 import eu.europeana.patternanalysis.view.ProblemPatternDescription;
 import eu.europeana.patternanalysis.view.RecordAnalysis;
 import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
 import java.util.HashSet;
 import java.util.List;
+import java.util.Map;
 import java.util.Objects;
 import java.util.Optional;
 import java.util.Set;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.regex.Pattern;
 import java.util.stream.Collectors;
 import org.apache.commons.collections.CollectionUtils;
 import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.text.similarity.LongestCommonSubsequence;
 
 /**
  * Class that contains functionality to analyze a record and retrieve all problem patterns.
  */
 public class ProblemPatternAnalyzer {
 
-  public static final int MIN_TITLE_LENGTH = 2;
+  private static final int MIN_TITLE_LENGTH = 2;
+  private static final int MAX_TITLE_LENGTH = 70;
+  private static final int MIN_DESCRIPTION_LENGTH = 50;
+  private static final int UNRECOGNIZABLE_CHARACTERS_THRESHOLD = 5;
+  private static final double LCS_CALCULATION_THRESHOLD = 0.9;
+  private static final int TITLE_DESCRIPTION_LENGTH_DISTANCE = 20;
+  // Match anything that is not alphanumeric in all languages or literal spaces. We cannot just use \\w
+  private static final String UNRECOGNIZABLE_CHARACTERS_REGEX = "[^\\p{L}\\p{M}\\p{N} ]";
```

Comment:
I find this like an overkill.
They are already regex categories by default, we are not adding anything by creating another naming here.
Perhaps I could use the full name of the category e.g. `\p{Letter}` `\p{Mark}` `\p{Number}`, would that suffice?
The space is necessary because we only want to capture literal space and not other type of invisible characters..

Reply:
I just realized that the full name category is not supported in code. I'll add a modified version of the suggestion.

Reply:
Okay I think I found a category that is self explanatory.

Reply:
yes :raised_hands: